### ISSUE TITLE
Make permuatations.json a subset of permuatations-large.json

### ISSUE THREAD

**Issue by shivankar-madaan** [bbb31/slurp#20]
Is it ok if we can have permuatation.json a complete subset of permuatation-large.json,so that when permuatation-large.json is run,none of the permutation is missed and we wont have to run it again with the smaller file.
There are some 335 words which are not present in the larger file,some of which are
```"es_%s.%s",
"emails_%s.%s",
"storage_%s.%s",
"ldap_%s.%s",
"%s_backups.%s",
"%s_main.%s",
"%s_svn.%s",
"pro_%s.%s",
"aws-logs_%s.%s",
"developer_%s.%s",
"bamboo_%s.%s",```

**Comment by bbb31** [comment]
Can you list the important ones? Also, which ones are not present in the larger one? Also looping in @jakewarren

**Comment by shivankar-madaan** [comment]
Ah ok .. that justifies it
But just for the heads up 
There are still buckets which exist with underscore with them.Although yes they are not anymore supported.

**Comment by jakewarren** [comment]
That is true, existing buckets would be grandfathered in. Others have stated in previous PRs that they have seen very low hit rates on buckets with underscores. I have not examined a significantly large amount of buckets but I've personally never found a single one with an underscore. 

I believe @menzow was working on collecting statistics on a large dataset so there may be hard data on the usage of the underscore character.

**Comment by bbb31** [comment]
I was thinking about this and potential solutions. I did not realize that buckets previously allowed underscores. My views on this is that while you cannot currently create buckets that have underscores, there still _may_ be buckets with underscores. For security reasons, _not_ including these gives a false sense of security for people who _think_ that this tool may not find their old underscore based buckets. Adding the underscores makes the permutations list grow longer and longer with no real benefit for currently created buckets. So, I think a potential solution to this problem would be to just have something like `permutations-underscore.json`; which would include these underscore versions. Thoughts?

**Comment by shivankar-madaan** [comment]
@bbb31 I think it is a cool idea.
But in my opinion having a separate file for underscores would be helpful,if your planning make a bigger list for it.
But for just 300-350 underscores permutations,I feel it would better fit in permuatations-large.json
Just and **only** a suggestion 😄 
Cheers.

**Comment by jakewarren** [comment]
@shivankar-madaan just a heads up, those 300 or so underscore permutations are only from the original wordlist - if you want to use underscore permutations there are several thousand that were not included during the creation of `permutations-large.json`. I still have an old commit for your reference: https://github.com/jakewarren/slurp/commit/d11376bf72c2c8870ea58346807978e1ee818084

**Comment by shivankar-madaan** [comment]
Oh ok,then we can have a separate file for underscores based on your call.
Also I had one doubt for a particular use case,for example
```
http://example-files.s3.amazonaws.com (is not accessible)
http://example-files.s3.amazonaws.com/users/ (is accessible)
```
So the files under the directory `users` would be accessible
Is there a way around that this can be checked?

**Comment by shivankar-madaan** [comment]
Any suggestion on the above query for checking w.r.t directories

**Comment by bbb31** [comment]
I agree that subdirectory enumeration would be pretty valuable, but it would also require another whole permutations list (one to find open buckets, one to probe them after). I do not oppose adding it, I just think it's out of the scope of this tool. I have seen a few forks that have "followed" found directories and downloaded everything that they've found.

For something like a specific target (domain or keywords targets) I can see that being useful, but if used with certstream it would literally run forever.

**Comment by bbb31** [comment]
This is a different issue than original thread; can you create a new issue?
